sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -523,6 +523,16 @@ object SimplifyConditionals extends Rule[LogicalPlan] with PredicateHelper {
         } else {
           e.copy(branches = branches.take(i).map(branch => (branch._1, elseValue)))
         }
+
+      case e @ EqualTo(c @ CaseWhen(branches, elseValue), right)
+          if c.deterministic &&
+            right.isInstanceOf[Literal] && branches.forall(_._2.isInstanceOf[Literal]) &&
+            elseValue.forall(_.isInstanceOf[Literal]) =>
+        if ((branches.map(_._2) ++ elseValue).forall(!_.equals(right))) {
```

Comment:
can we use an `EqualTo` expression to compare literals? and how about the null semantic?